core/src/main/scala/org/broadinstitute/dsde/rawls/entities/base/ExpressionEvaluationSupport.scala

```diff
@@ -24,9 +24,42 @@ trait ExpressionEvaluationSupport {
     valuesByEntity.map({ case (entityName, values) => SubmissionValidationEntityInputs(entityName, values.toSet) }).toStream
   }
 
+  protected def isStringInputType(input: MethodInput): Boolean = {
+    val valuetype = input.workflowInput.getValueType
+
+    // TODO: map types? tuple types?
+    valuetype.getTypeName == TypeNameEnum.STRING ||
+      (valuetype.getTypeName == TypeNameEnum.OPTIONAL &&
+        valuetype.getOptionalType.getTypeName == TypeNameEnum.STRING) ||
+      (valuetype.getTypeName == TypeNameEnum.ARRAY &&
+        valuetype.getArrayType.getTypeName == TypeNameEnum.STRING)
+  }
```

Comment:
I have reached out to Batch to ask about map and tuple types, perhaps we need to handle those too.

Reply:
I don't think it's _too_ likely that users are turning strings into native composite types in their WDLs.

When I imagine them using ints or bools as strings, I think stuff like this:

`reticulate_splines` = `"true"`
`memory_gb` = `"4"`
```
command <<<
    java -jar my_tool.jar -Xmx~{memory_gb}g -Dreticulate_splines=~{reticulate_splines}
>>>
```

Which is to say, placing ints or bools in places where they would get cast to strings anyway.

If they were going to write a WDL that takes in a map as a string, and then really uses it as a map, they would need to introduce an explicit [`read_map()`](https://github.com/openwdl/wdl/blob/main/versions/1.0/SPEC.md#mapstring-string-read_mapstringfile) call that is surely more effort than having the input be a `Map` in the first place.

That said, this is only in my imagination, and any firsthand experience with how users interact with this behavior is helpful!

Reply:
Also, the Terra data table isn't going to automatically try to cast stringified maps or tuples to actual tuples, is it?

i.e. `"{"a": "b"}"` to `{"a": "b"}`

If no, then there is no need to _reverse_ that conversion for compatibility on the backend, when we go to submit a workflow.

Reply:
no, the data table doesn't do any de-stringification. Thank you for the feedback here! I'm going to leave the PR as-is … and of course if we hear complaints/requests later, we can always make another PR.

Reply:
Sounds good! And, thanks for this valuable work.